src/java.desktop/unix/native/libawt_xawt/awt/screencast_pipewire.c

```diff
@@ -0,0 +1,915 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#ifdef HEADLESS
+#error This file should not be included in headless library
+#endif
+
+#include <dlfcn.h>
+#include "jni_util.h"
+#include "awt.h"
+#include "screencast_pipewire.h"
+#include "fp_pipewire.h"
+#include <stdio.h>
+
+#include "gtk_interface.h"
+#include "gtk3_interface.h"
+
+int DEBUG_SCREENCAST_ENABLED = FALSE;
+
+#define EXCEPTION_CHECK_DESCRIBE_CLEAR() if ((*env)->ExceptionCheck(env)) { \
+                                            (*env)->ExceptionDescribe(env); \
```

Comment:
Describe already clears the exception
https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#exceptiondescribe

Reply:
Updated.